### DISCUSSION TITLE
custom package development 404 on package view

### DISCUSSION THREAD

**Discussion by sts-ryan-holton** [laravel/framework#34520] [General]
- Laravel Version: 8.x
- PHP Version: 7.4.x
- Database Driver & Version: N/A

### Description:

I've just starting developing a Laravel 8 package that contains some custom routes and a view. I'm trying to load this package into my main Laravel project and access the configured route from my package, I'm new to developing packages in Laravel and despite following the documentation I'm getting a 404 not found. I've ran `php artisan config:cache`, and `php artisan route:list` to see whether my package's route is at least showing, and it is, but is failing to load the view.

My package's view is listed under `/src/views/form.blade.php` and my route is `/form` within the package, but accessing this in my project doesn't load the view from the package...

My package name is called: **TestPackage**

Not sure where I'm going wrong, some guidance would be appreciated here, the contents of `php artisan route:list` gives me from my main project:

```bash
+--------+----------+----------+------+---------+------------+
| Domain | Method   | URI      | Name | Action  | Middleware |
+--------+----------+----------+------+---------+------------+
|        | GET|HEAD | /        |      | Closure | web        |
|        | GET|HEAD | api/user |      | Closure | api        |
|        |          |          |      |         | auth:api   |
|        | GET|HEAD | form     |      | Closure | web        |
+--------+----------+----------+------+---------+------------+
```

My package's `routes.php` is:

```php
Route::group(['namespace'=>'company\TestPackage\Http\Controllers'], function() {

  // Attempt 1
  // Route::get('form','TestPackageController@renderForm')->name('test-package');

  // Attempt 2, ideal
  Route::get('form', function () {
    return view('test-package::form');
  });

});
```

And my service provider is...

```php
<?php

namespace company\TestPackage;

use Illuminate\Support\ServiceProvider;

class TestPackageServiceProvider extends ServiceProvider
{
  public function boot() {
    $this->loadRoutesFrom(__DIR__.'/routes/web.php');
    $this->loadViewsFrom(__DIR__.'/views', 'test-package');
  }

  public function register() {

  }
};
```

Thanks!

**Comment by driesvints** [comment]
Hi there,

Thanks for reporting but it looks like this is a question which can be asked on a support channel. Please only use this issue tracker for reporting bugs with the library itself. 

You can post support questions on our discussions tab instead so I'm moving it over there. I won't be able to help you out personally but others who are watching this repository can help you out.

Alternatively, you can post this question on one of the following channels:

- [Laracasts Forums](https://laracasts.com/discuss)
- [Laravel.io Forums](https://laravel.io/forum)
- [StackOverflow](https://stackoverflow.com/questions/tagged/laravel)
- [Discord](https://discordapp.com/invite/KxwQuKb)
- [Larachat](https://larachat.co)
- [IRC](https://webchat.freenode.net/?nick=laravelnewbie&channels=%23laravel&prompt=1)

Thanks.
